src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/AttributeData.php

```diff
@@ -154,12 +153,7 @@ private function addRelationData(&$parentData, $childAttributes, $relation)
         );
 
         if (isset($relation['configurable_attributes']) && !empty($relation['configurable_attributes'])) {
-            $addedChildrenAttributes = array_diff(
-                $childAttributeCodes,
-                $this->forbidenChildrenAttributeCode,
```

Comment:
Why removing the `forbidenChildrenAttributeCode` ?

Reply:
Because found that same (without $parentAttributeCodes) array_diff exists on 152 line.

Reply:
There are indeed two problems :
- if `$childrenAttributes` is no longer for a configurable only the attributes that exist solely in the children products, then 
- - it makes sense to remove the historic `array_diff` of line 157 to 161
- - but it would make also sense to remove the following `array_merge` of line 162 : I don't see the point of re-merging `$parentData['children_attributes']` into `$childrenAttributes`
- as stated by Romain in that comment https://github.com/Smile-SA/elasticsuite/pull/1545#issuecomment-541676432, now the `name` of children products are indexed into the parent, which might totally mess sorting products by name for configurables
- - so 'name' should be added to '$forbidenChildrenAttributeCode'
- - but what about bundle products ? don't they suffer **today** about that potentiel "sort by name" issue ?

I'm yet to fully understand the original issue (I haven't tried to reproduce it), but on a hunch, I think it's due to different expectations between (non-visible) children products made out of a configurable product vs children products existing in their own right handpicked to form a configurable product.

At this point, I would cover all bases having this new behavior being configurable via an admin option which is probably a bad idea.